biosys/apps/main/models.py

```diff
@@ -342,7 +347,14 @@ def record_queryset(self):
 
     @property
     def record_count(self):
-        return self.record_model.objects.filter(dataset=self).count()
+        return self.record_queryset.count()
+
+    @property
+    def extent(self):
+        if self.type != Dataset.TYPE_GENERIC:
```

Comment:
in prevision of the  removal of the dataset type you should remove this test and add a filter in the queryset
```
self.record_queryset.exclude(geometry__isnull=True).....
```
or maybe the Extent function does it